staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go

```diff
@@ -51,35 +48,28 @@ const (
 	String             // The IntOrString holds a string.
 )
 
-// FromInt creates an IntOrString object with an int32 value. It is
-// your responsibility not to call this method with a value greater
-// than int32.
-// Deprecated: use FromInt32 instead.
-func FromInt(val int) IntOrString {
-	if val > math.MaxInt32 || val < math.MinInt32 {
-		klog.Errorf("value: %d overflows int32\n%s\n", val, debug.Stack())
-	}
-	return IntOrString{Type: Int, IntVal: int32(val)}
+// FromInt creates an IntOrString object with an int32 value.
+func FromInt(val int32) IntOrString {
+	return IntOrString{Type: Int, IntVal: val}
```

Comment:
i don't have the full context on these changes, but i find the naming a bit misleading. `FromInt` presumably would take an `int` or the system register size. while `FromInt32` would take the `int32`.

this constructs an `IntOrString` tuple which also is a bit misleading as the `Int` is of size 32bit.
also, when the user would like to obtain the `Int` from that tuple they would get a `int` (system size) via `IntValue() int` instead of the storage size of 32bit.

Reply:
Apologies, I meant to add the context in the PR description. This is based on comments on #117631.

The `IntOrString` tuple is indeed confusing, and there’s a todo in the code to rename it to `Int32OrString`; I haven’t attempted to address that. I noticed the `IntValue()` confusion as well, it would probably make sense to change that to return an `int32` so that the output matches the input.